app/src/main/java/org/fossasia/pslab/communication/ScienceLab.java

```diff
@@ -333,7 +333,30 @@ private void runInitSequence(Boolean loadCalibrationData) throws IOException {
                             }
                         }
 
-                        // todo : port remaining function
+                        ArrayUtils.reverse(fitVals);
+                        PolynomialFunction fitFunction = new PolynomialFunction(fitVals);
+                        double[] yData = new double[DACX.length];
+                        for (int j = 0; j < DACX.length; j++) {
+                            yData[j] = fitFunction.value(DACX[j]) - (OFF.get(j) * slope + intercept);
+                        }
+                        int LOOKBEHIND = 100;
+                        int LOOKAHEAD = 100;
```

Comment:
Yeah, it might have some significance in terms of calibrating the raw values. Yes we can assign a common variable and use it, currently we are taking 100 values from both ahead and back of current value. That might have yielded best results. Rest @jithinbp can clear as it was like this in python-communication-library.